evap/results/tools.py

```diff
@@ -171,42 +170,109 @@ def _calculate_results_impl(course):
+        grade_questions_distribution = avg_distribution([result.counts for result in grade_question_results])
+        non_grade_question_results = [result for result in grouped_results[contributor] if result.question.is_non_grade_rating_question]
+        non_grade_questions_average = avg([result.average for result in non_grade_question_results])
+        non_grade_questions_distribution = avg_distribution([result.counts for result in non_grade_question_results])
+        if contributor:
+            # collect the grade question average, non-grade question average and the distributions for each contributor
+            contributor_averages.append(mix(grade_questions_average, non_grade_questions_average, settings.CONTRIBUTOR_GRADE_QUESTIONS_PERCENTAGE))
+            contributor_distributions.append(weighted_avg_distribution(
+                [grade_questions_distribution, non_grade_questions_distribution],
+                [settings.CONTRIBUTOR_GRADE_QUESTIONS_PERCENTAGE, 1 - settings.CONTRIBUTOR_GRADE_QUESTIONS_PERCENTAGE]
+            ))
+        else:
+            # calculate the grade question average, non-grade question average and the distributions for the course
+            course_grade_questions_average = grade_questions_average
+            course_grade_questions_distribution = grade_questions_distribution
+            course_non_grade_questions_average = non_grade_questions_average
+            course_non_grade_questions_distribution = non_grade_questions_distribution
+
+    if course.is_single_result:
+        course_grade_questions_average = None
+        course_non_grade_questions_average = None
+
+    # calculate the average grade and distribution over all contributors
+    contributors_average = avg(contributor_averages)
+    contributors_distribution = avg_distribution(contributor_distributions)
+
+    # calculate to which percentage each group influences the total grade
+    contributions_percentage = settings.CONTRIBUTIONS_PERCENTAGE if contributors_average is not None else 0
+    course_grade_questions_percentage = settings.COURSE_GRADE_QUESTIONS_PERCENTAGE if course_grade_questions_average is not None else 0
+    course_non_grade_questions_percentage = 1 - settings.CONTRIBUTIONS_PERCENTAGE - settings.COURSE_GRADE_QUESTIONS_PERCENTAGE if course_non_grade_questions_average is not None else 0
+
+    scale = 1 / sum([contributions_percentage, course_grade_questions_percentage, course_non_grade_questions_percentage])
+    contributions_percentage *= scale
+    course_grade_questions_percentage *= scale
+    course_non_grade_questions_percentage *= scale
+
+    # calculate average grade
+    average_grade = contributions_percentage * (contributors_average or 0) + \
+                    course_grade_questions_percentage * (course_grade_questions_average or 0) + \
+                    course_non_grade_questions_percentage * (course_non_grade_questions_average or 0)
+
+    # calculate average distribution
```

Comment:
this comment and the one above don't really add anything in my opinion since  the code quite clearly says `average distribution = some calculation`

Reply:
no they don't but the grouping would be off if some parts wouldn't have a comment. still remove them?

Reply:
it won't be off if we remove more comments ;) in general i would use text in comments primarily for commenting and not grouping.